uikit/positionlockfab/viewmodel/src/main/java/com/sygic/ui/viewmodel/positionlockfab/PositionLockFabViewModel.kt

```diff
@@ -100,55 +100,8 @@ class PositionLockFabViewModel internal constructor(
         cameraModel.zoomLevel = zoomLevel
     }
 
-    private fun setNextLockState(block: () -> Unit) {
-        checkPermission {
-            checkLocationEnabled(block)
-        }
-    }
-
-    private fun checkPermission(onSuccess: () -> Unit) {
-        permissionsManager.checkPermissionGranted(
-            Manifest.permission.ACCESS_FINE_LOCATION,
-            Observer { permissionEnabled ->
-                if (!permissionEnabled) {
-                    permissionsManager.requestPermission(Manifest.permission.ACCESS_FINE_LOCATION,
-                        object : PermissionsManager.PermissionCallback {
-                            override fun onPermissionGranted(permission: String) {
-                                SygicEngine.openGpsConnection()
-                                onSuccess()
-                            }
-
-                            override fun onPermissionDenied(permission: String) {
-                                /* Currently do nothing */
-                            }
-                        })
-                } else {
-                    onSuccess()
-                }
-            })
-    }
-
-    private fun checkLocationEnabled(onSuccess: () -> Unit) {
-        locationManager.checkGpsEnabled(Observer { gpsEnabled ->
-            if (!gpsEnabled) {
-                locationManager.requestToEnableGps(object : LocationManager.EnableGpsCallback {
-                    override fun onResult(@EnableGpsResult result: Int) {
-                        when (result) {
-                            EnableGpsResult.ENABLED -> onSuccess()
-                            EnableGpsResult.DENIED -> {
-                                /* Currently do nothing */
-                            }
-                        }
-                    }
-                })
-            } else {
-                onSuccess()
-            }
-        })
-    }
-
     override fun onStop(owner: LifecycleOwner) {
-        cameraModel.addModeChangedListener(this)
-        PositionManager.getInstance().stopPositionUpdating() //ToDo: MS-4555
+        cameraModel.removeModeChangedListener(this)
+        locationManager.positionOnMapEnabled = false
```

Comment:
nie je toto problem, ze nastavis `positionEnabled` `false`, ale user to mohol zapnut cez BrowseMapFragment a nebude vediet, preco je to vypnute? Ako je to `onStop` len ci sa nenarusi ta konzistentnost...

Reply:
Pravdu máš, upravím,